File: Python/Stacks/Stack.py
class Node:
    def __init__(self,val):
        self.val = val
        self.next = None

class Stack:
    def __init__(self):
        self.first = None
        self.last = None
        self.size = 0

    def push(self, val):
        newNode = Node(val)
        if self.size == 0:
            self.first = newNode
            self.last = newNode
        else:
            temp = self.first
            self.first = newNode
            newNode.next = temp
        self.size += 1
        return self

    def pop(self):
        if self.size == 0:
            return None
        popped = self.first
        if self.size == 1:
            self.first = None
            self.last = None
        else:
            newHead = popped.next
            popped.next = None
            self.first = newHead
        self.size -= 1
        return popped.val

    def print(self):
        output = []
        if self.size != 0:
            current = self.first
            output.append(current.val)
            while current.next != None:
                current = current.next
                output.append(current.val)
        return output

    def getSize(self):
        return self.size
            
        
stack = Stack()
stack.push(5)
print(stack.print())
stack.push(6)
print(stack.print())
stack.push(7)
print(stack.print())
stack.push(8)
print(stack.print())
print(stack.pop())
print(stack.print())
print(stack.pop())
print(stack.print())
print(stack.pop())
print(stack.print())
print(stack.pop())
print(stack.print())

File: JavaScript/Stacks/Stack.js
class Node {
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}

class Stack {
    constructor() {
        this.first = null;
        this.last = null;
        this.size = 0;
    }

    push(val) {
        let newNode = new Node(val);
        if (this.size === 0) {
            this.first = newNode;
            this.last = newNode;
        } else {
            let first = this.first;
            newNode.next = first;
            this.first = newNode;
        }
        this.size += 1;
        return this.size;
    }

    pop() {
        if (this.size === 0) {
            return;
        } else if (this.size === 1) {
            var temp = this.first;
            this.first = null;
            this.last = null;
        } else {
            var temp = this.first;
            this.first = temp.next;
            temp.next = null;
        }
        this.size -= 1;
        return temp.val;
    }
}

let stack = new Stack()

console.log(stack.push('Hi'))
console.log(stack.push('There'))
console.log(stack.push('How'))
console.log(stack.push('Are You?'))
console.log(stack)
console.log(stack.pop())
console.log(stack.pop())
console.log(stack.pop())
console.log(stack.pop())
console.log(stack.pop())